### PR TITLE
Created a small package with a helper class for creating PageObjects.

### PR DESCRIPTION
**Pull request by fatherOfLegends**

### CODE REVIEW

File: packages/page_object/lib/src/page_object.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+/// This class exists so page objects can seamlessly be passed to widget testing APIs as Finders. This helps
+/// to remove code noise from tests. See the tests in this package for an example.
+class PageObject extends Finder {
```
Comment (coreysprague) [inline packages/page_object/lib/src/page_object.dart:6]: So, with this... the generic GWT Harness... and presumably a set of GWT assertions to wrap widget tester...

how micro do we want to on these packages? I'm of two minds:

1) bundle them together
2) package them separately, but then offer a composite package that bundles them together?

I'm just not convinced at how much utility they provide individually? thoughts?

Comment (fatherOfLegends) [reply packages/page_object/lib/src/page_object.dart:6]: While we use all of these tools together I view them as completely separate functionality that other teams might only want pieces of. The friction to maintain them and use them separately is very low. If we wanted to create a flutter glove box package that includes all of these I would be open to that.

Regarding the collection GWT assertions, I could see other teams wanting very different names or similar names but different signatures. I'd prefer not to be overly opinionated about that API.

File: packages/page_object/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# pedantic has a strong foundation, but right now they only enable a small set of lints
```
Comment (coreysprague) [inline packages/page_object/analysis_options.yaml:1]: minor (not for this PR), but I wonder if we can move the shared analysis file to the root of the repo and reference it via ../../

then we don't have to duplicate per package.

can be a future optimization though

File: packages/page_object/test/page_object_example_usage_test.dart [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:page_object/page_object.dart';
+
+void main() {
```
Comment (coreysprague) [inline packages/page_object/test/page_object_example_usage_test.dart:5]: future note... might be worth an example that shows composition (where one FinderWrapper holds a reference to another FinderWrapper)

**Review by coreysprague — APPROVED** [review]
👍

File: packages/page_object/analysis_options.yaml (hunk 2 above)
Comment (fatherOfLegends) [reply packages/page_object/analysis_options.yaml:1]: Ya this is a good idea.

Comment (fatherOfLegends) [reply packages/page_object/analysis_options.yaml:1]: #92

File: packages/page_object/test/page_object_example_usage_test.dart (hunk 3 above)
Comment (fatherOfLegends) [reply packages/page_object/test/page_object_example_usage_test.dart:5]: This is a good idea, I will attempt to incorporate it.

Comment (fatherOfLegends) [reply packages/page_object/test/page_object_example_usage_test.dart:5]: done

**Pull request merged**